File: eCommerce/eCommerce.Services/Database/PeerChallenge.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace eCommerce.Services.Database
{
    public class PeerChallenge
    {
        public int PeerChallengeId { get; set; }
        public int IzazivacId { get; set; }
        public User Izazivac { get; set; }
        public int IzazvaniId { get; set; }
        public User Izazvani { get; set; }
        public int ChallengeId { get; set; }
        public Challenge Challenge { get; set; }

        public StatusPeer StatusPeera { get; set; }
        public DateTime DatumKreiranja { get; set; }
        public enum StatusPeer
        {
            Poslan=1,
            Prihvacen=2,
            Odbijen=3
        }
    }
}

File: eCommerce/UI/ecommerce_mobile/lib/model/peer_challenge.dart
import 'package:ecommerce_mobile/model/challenge.dart';
import 'package:ecommerce_mobile/model/user.dart';
import 'package:json_annotation/json_annotation.dart';

part 'peer_challenge.g.dart';

@JsonSerializable()
class PeerChallenge {
  int? peerChallengeId;
  int? izazivacId;
  User? izazivac;
  int? izazvaniId;
  User? izazvani;
  int? challengeId;
  Challenge? challenge;
  String? statusPeera;
  DateTime? datumKreiranja;

  PeerChallenge({
   this.peerChallengeId,
   this.izazivacId,
   this.izazivac,
   this.izazvaniId,
   this.izazvani,
   this.challengeId,
   this.challenge,
   this.statusPeera,
   this.datumKreiranja,
  });

  factory PeerChallenge.fromJson(Map<String, dynamic> json) => _$PeerChallengeFromJson(json);

  Map<String, dynamic> toJson() => _$PeerChallengeToJson(this);
} 